data/result-exporter/swe-cne-exporter/src/main/java/com/farao_community/farao/data/swe_cne_exporter/SweAdditionalConstraintSeriesCreator.java

```diff
@@ -0,0 +1,83 @@
+import com.farao_community.farao.data.crac_api.Instant;
+import com.farao_community.farao.data.crac_api.cnec.AngleCnec;
+import com.farao_community.farao.data.crac_creation.creator.cim.crac_creator.CimCracCreationContext;
+import com.farao_community.farao.data.crac_creation.creator.cim.crac_creator.cnec.AngleCnecCreationContext;
+import com.farao_community.farao.data.swe_cne_exporter.xsd.AdditionalConstraintSeries;
+
+import java.math.BigDecimal;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static com.farao_community.farao.commons.logs.FaraoLoggerProvider.BUSINESS_WARNS;
+
+/**
+ * Generates AdditionalConstraintSeries for SWE CNE format
+ *
+ * @author Godelaine de Montmorillon {@literal <godelaine.demontmorillon at rte-france.com>}
+ */
+public class SweAdditionalConstraintSeriesCreator {
+    private final SweCneHelper sweCneHelper;
+    private final CimCracCreationContext cracCreationContext;
+
+    public SweAdditionalConstraintSeriesCreator(SweCneHelper sweCneHelper, CimCracCreationContext cracCreationContext) {
+        this.sweCneHelper = sweCneHelper;
+        this.cracCreationContext = cracCreationContext;
+    }
+
+    public List<AdditionalConstraintSeries> generateAdditionalConstraintSeries(Contingency contingency) {
+        List<AdditionalConstraintSeries> additionalConstraintSeriesList = new ArrayList<>();
+        if (Objects.isNull(sweCneHelper.getAngleMonitoringResult())) {
+            return additionalConstraintSeriesList;
+        }
+        List<AngleCnecCreationContext> sortedAngleCnecs = cracCreationContext.getAngleCnecCreationContexts().stream()
+                .filter(AngleCnecCreationContext::isImported)
+                .sorted(Comparator.comparing(AngleCnecCreationContext::getNativeId))
+                .collect(Collectors.toList());
+        if (Objects.isNull(contingency)) {
+            List<AngleCnecCreationContext> prevAngleCnecs = sortedAngleCnecs.stream().filter(angleCnecCreationContext -> Objects.isNull(angleCnecCreationContext.getContingencyId()))
+                    .collect(Collectors.toList());
+            prevAngleCnecs.forEach(angleCnecCreationContext ->
+                BUSINESS_WARNS.warn("Preventive angle cnec {} will not be added to CNE file", angleCnecCreationContext.getNativeId()));
+        } else {
+            List<AngleCnecCreationContext> contingencyAngleCnecs = sortedAngleCnecs.stream()
+                    .filter(angleCnecCreationContext -> angleCnecCreationContext.getContingencyId().equals(contingency.getId()))
+                    .collect(Collectors.toList());
+            contingencyAngleCnecs.forEach(angleCnecCreationContext -> {
+                AdditionalConstraintSeries additionalConstraintSeries = generateAdditionalConstraintSeries(angleCnecCreationContext);
+                if (Objects.nonNull(additionalConstraintSeries)) {
+                    additionalConstraintSeriesList.add(additionalConstraintSeries);
+                }
+            });
```

Comment:
you can use
stream.filter(contingency).map(generateAdditional..).filter(nonNull).foreach(additionalConstraintSeriesList::add)